test/parallel/test-fs-utimes.js

```diff
@@ -164,7 +164,26 @@ runTest(new Date('1982-09-10 13:37'), new Date('1982-09-10 13:37'), function() {
 });
 
 
+// Ref: https://github.com/nodejs/node/issues/13255
+if (common.isWindows) {
+  common.refreshTempDir();
+  const path = `${common.tmpDir}/test-utimes-precision`;
+  fs.writeFileSync(path, '');
+
+  // this value would get converted to (double)1713037251359.9998
+  const truncate_mtime = 1713037251360;
+  fs.utimesSync(path, truncate_mtime / 1000, truncate_mtime / 1000);
+  const truncate_stats = fs.statSync(path);
+  assert.strictEqual(truncate_mtime, truncate_stats.mtime.getTime());
+
+  // if this value is cast to `signed long` it gets converted to -2135622133469
```

Comment:
I added to the comment why it's not a problem in POSIX, because there timestamps are stored in two fields `struct {long t_sec, long t_usec}`, and neither can overflow.
The problem is just how we convert from the NTFS `uint64_t` that hold nanoseconds, back to seconds.
IMHO This change is safe, since even the numbers from NTFS if properly converted to seconds fit in a `long` and in a `double`.  (notice that the actual call to `utimesSync` divides these numbers by 1000). Also no FS will give us negative times, and the paper mentioned in the bug ([interpretation-of-ntfs-timestamps](https://articles.forensicfocus.com/2013/04/06/interpretation-of-ntfs-timestamps/)), says we just need to interpret it right.

You might be right in that the calculations in https://github.com/nodejs/node/blob/master/deps/uv/src/win/fs.c#L91 should be done with `uint`